tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/ControlFlowDebugTest.java

```diff
@@ -0,0 +1,165 @@
+        resumeProgram(debugHitInfo.getRight(), DebugResumeKind.STEP_OVER);
+        debugHitInfo = waitForDebugHit(10000);
+        Assert.assertEquals(debugHitInfo.getLeft(), new BallerinaTestDebugPoint(testEntryFilePath, 39));
+
+        // Test for debug engage in 'match' statement
+        resumeProgram(debugHitInfo.getRight(), DebugResumeKind.NEXT_BREAKPOINT);
+        debugHitInfo = waitForDebugHit(10000);
+        Assert.assertEquals(debugHitInfo.getLeft(), testBreakpoints.get(5));
+
+        // Test for debug engage inside 'match' statement when condition is true.
+        resumeProgram(debugHitInfo.getRight(), DebugResumeKind.STEP_OVER);
+        debugHitInfo = waitForDebugHit(10000);
+        Assert.assertEquals(debugHitInfo.getLeft(), new BallerinaTestDebugPoint(testEntryFilePath, 47));
+
+        resumeProgram(debugHitInfo.getRight(), DebugResumeKind.STEP_OVER);
+        debugHitInfo = waitForDebugHit(10000);
+        Assert.assertEquals(debugHitInfo.getLeft(), new BallerinaTestDebugPoint(testEntryFilePath, 48));
+
+        // Test for debug engage in lambda - iterable arrow operation
+        resumeProgram(debugHitInfo.getRight(), DebugResumeKind.NEXT_BREAKPOINT);
+        debugHitInfo = waitForDebugHit(10000);
+        Assert.assertEquals(debugHitInfo.getLeft(), testBreakpoints.get(6));
+
+        // Test for debug engage in Asynchronous function call - Non-blocking calls
+        resumeProgram(debugHitInfo.getRight(), DebugResumeKind.NEXT_BREAKPOINT);
+        debugHitInfo = waitForDebugHit(10000);
+        Assert.assertEquals(debugHitInfo.getLeft(), testBreakpoints.get(7));
+
+        // Prepare variables for visibility test by adding a debug point at the end of the .bal file.
+        resumeProgram(debugHitInfo.getRight(), DebugResumeKind.NEXT_BREAKPOINT);
+        debugHitInfo = waitForDebugHit(10000);
+
+        Variable[] variables = fetchDebugHitVariables(debugHitInfo.getRight());
+        Arrays.sort(variables, compareByName);
+
+        // Variable visibility test inside 'match' statement
+        assertVariable(variables[6], "v07_intVar", "7", "int");
+
+        // Variable visibility test for lambda - iterable arrow operation
+        assertVariable(variables[8], "v09_animals", "map", "map");
+
+        // Variable visibility test for lambda child variables
+        Variable[] lamdaChildVariables = getChildVariable(variables[8]);
+        Arrays.sort(lamdaChildVariables, compareByName);
+        assertVariable(lamdaChildVariables[0], "a", "ANT", "string");
+
+        // Variable visibility test for Asynchronous function call (Non-blocking calls)
+        assertVariable(variables[9], "v10_future", "future", "future");
+
+        // Variable visibility test for Asynchronous function call child variables
+        Variable[] asyncChildVariables = getChildVariable(variables[9]);
+        Arrays.sort(lamdaChildVariables, compareByName);
+        assertVariable(asyncChildVariables[1], "result", "90", "int");
+    }
```

Comment:
Is there any special reason for doing variable tests in here? We have separate tests for the variable validation right? Sorry If I've missed anything.

Reply:
This is to verify variable name, type and values are properly visible inside control flows. Previous variable visibility tests do not cover this scenario.